src/java/org/apache/cassandra/auth/MutualTlsAuthenticator.java

```diff
@@ -208,9 +224,29 @@ public AuthenticatedUser getAuthenticatedUser() throws AuthenticationException
                 nospamLogger.error(msg, identity);
                 throw new AuthenticationException(MessageFormatter.format(msg, identity).getMessage());
             }
+
+            // Validates that the certificate validity period does not exceed the maximum certificate configured validity period
+            int minutesToCertificateExpiration = certificateValidityPeriodValidator.validate(clientCertificateChain);
+            int daysToCertificateExpiration = toDays(minutesToCertificateExpiration);
+
+            if (certificateValidityWarnThreshold != null
+                && minutesToCertificateExpiration < certificateValidityWarnThreshold.toMinutes())
+            {
+                nospamLogger.warn("Certificate with identity '{}' will expire in {} minutes",
+                                  identity, minutesToCertificateExpiration);
+            }
+
+            // Report metrics on client certificate expiration
+            MutualTlsMetrics.instance.clientCertificateExpirationDays.update(daysToCertificateExpiration);
+
             return new AuthenticatedUser(role, MTLS, Collections.singletonMap(METADATA_IDENTITY_KEY, identity));
         }
 
+        private int toDays(int minutes)
+        {
+            return (int) TimeUnit.MINUTES.toDays(minutes);
```

Comment:
just realized, something like 30 minutes will convert to 0 days, which may lead someone to believe that they have a certificate that is expired when they are really close on the margin.  It's probably fine but could see that possibly be misleading.

Reply:
done